vars/buildImage.groovy

```diff
@@ -0,0 +1,99 @@
+// Create a Job and BuildConfig Template to trigger the Build
+// Available parameters:
+//    dockerFile    string -- DockerFile used for the buildconfig
+//    gitURL:       string -- Git URL used for the buildconfig
+//    image         string -- Image used for the buildconfig
+//    nameSpace:    string -- Namespace name to run the build
+def call(params = [:]) {
+    def imageName
+    pipeline {
+        // We don't need any specify agent just to add the buildconfig and trigger it,
+        // this pod will be destroyed in the end
+        agent {
+            node { label 'nodejs' }
+        }
+        stages {
+            stage('Prepare Env') {
+                steps{
+                   script{
+                       openshift.withCluster() {
+                           openshift.withProject() {
+
+                               def bcJSON  = libraryResource 'com/github/coreos/buildconfig.json'
+                               def jobJSON = libraryResource 'com/github/coreos/job.json'
+                               def jobObj  = readJSON text: jobJSON
+                               def bcObj   = readJSON text: bcJSON
+                               def tagName = UUID.randomUUID()
+
+                               jobObj['parameters'][0] +=['name': 'NAME', 'value': "cleanup-${tagName}".toString()]
+
+                               writeYaml(file: "job-${tagName}.yaml", data: jobObj, overwrite: true)
+                               jobTemplatePath = env.WORKSPACE+"/job-${tagName}.yaml"
```

Comment:
If I'm reading https://github.com/openshift/jenkins-client-plugin#cannot-live-without-openshift-templates-no-problem right, I think we can avoid writing to disk entirely and just instantiate the template via `openshift.create(openshift.process(...))`.

Reply:
Indeed!!. Much better =)